org.scala-ide.sdt.core.tests/src/org/scalaide/core/testsetup/Bdd.scala

```diff
@@ -0,0 +1,55 @@
+package org.scalaide.core.testsetup
+
+import java.io.File
+
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.resources.IncrementalProjectBuilder
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.eclipse.jdt.core.IJavaProject
+import org.junit.Assert
+import org.scalaide.core.IScalaProject
+import org.scalaide.util.eclipse.EclipseUtils
+
+import SDTTestUtils.enableAutoBuild
+import SDTTestUtils.getErrorMessages
+import SDTTestUtils.sourceWorkspaceLoc
+import SDTTestUtils.workspace
+
+/**
+ * Behavioral Driven Development methods.
+ */
+trait Bdd {
+  import SDTTestUtils._
+
+  def givenCleanWorkspaceForProjects(projects: IProject*): Unit = {
+    workspace.build(IncrementalProjectBuilder.CLEAN_BUILD, new NullProgressMonitor)
+    projects.foreach { project =>
+      Assert.assertTrue(getErrorMessages(project).isEmpty)
+    }
+  }
+}
+
+object Bdd {
+  implicit def jProjectToIProject(project: IJavaProject): IProject = project.getProject
```

Comment:
Ok, I checked usages of those and it seems sensible. Problem is we have several helper methods that requires `IProject`. Unfortunately both`IJavaProject` and `IScalaProject` encapsulates it, not extends.

With those implicits one can just pass `jProject: IJavaProject` to method that requires `IProject`.

One thing I would change here is rename `objecy Bdd` into `trait IProjectHelpers` and mix it to some base integration test.